persistence/sql/persister_session.go

```diff
@@ -82,15 +83,34 @@ func (p *Persister) UpsertSession(ctx context.Context, s *session.Session) error
 
 	s.NID = p.NetworkID(ctx)
 
-	if err := p.Connection(ctx).Find(new(session.Session), s.ID); errors.Is(err, sql.ErrNoRows) {
-		// This must not be eager or identities will be created / updated
-		return errors.WithStack(p.GetConnection(ctx).Create(s))
-	} else if err != nil {
-		return errors.WithStack(err)
-	}
+	return errors.WithStack(p.Transaction(ctx, func(ctx context.Context, tx *pop.Connection) error {
+		if err := tx.Find(new(session.Session), s.ID); errors.Is(err, sql.ErrNoRows) {
```

Comment:
This needs to respect `nid`, it currently doesn't.

Can you also please use `sqlcon.HandleError` and `sqlcon.ErrNotFound()`